### ISSUE TITLE
Connecting to Storage Account using SAS string with a trailing blank causes delete entity operations to fail

### ISSUE THREAD

**Issue by KerbalSpike** [microsoft/AzureStorageExplorer#7680]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.32.0

### Architecture

x64

### Storage Explorer Build Number

20231114.10

### Platform

Windows

### OS Version

Windows 11 Version 22H2 (Build 22621.3007)

### Bug Description

I have a certain storage account where I'm unable to delete any entities from any table using the Azure Storage Explorer. Attempting to do so always results in a "RestError" with the message "The specified resource name length is not within the permissible limits."

I didn't understand how that could be the case, since the resource was clearly there, I had full CRUD access from my app (including delete) and Azure Storage Explorer itself had no problem fetching as well as updating and inserting data. Only delete operations issued from Azure Storage Explorer ran into this error.

After a lot of searching, digging and testing I figured out that the root cause for this problem was an SAS string, that accidentally included a trailing blank at the end, that gets interpreted as belonging to the table endpoint URL.

This can easily be spotted by selecting  table in the account and look at the URL in the properties section. It will look something like
`https://{AccountName}.table.core.windows.net/ /tablename`

For some reason and very confusingly, this only affects the delete operation. Table data can be viewed, edited and inserted without issues.

### Steps to Reproduce

1. Launch Storage Explorer
2. Connect to a valid Storage Account using an SAS string that
    * has the TableEndpoint property at the end of the string
    * includes a trailing space
3. Select a table (create if none exists)
4. Select at least one entity in the table (create if none exists)
5. Click Delete and confirm the action in the subsequent dialog.

### Actual Experience

Operation fails with error
`Failed to delete entities in table 'tablename'`
Error Details json includes:  
```
"name": "RestError",
"message": "The specified resource name length is not within the permissible limits. [snip]"
```

### Expected Experience

Selected entity/entities is/are deleted.

**Comment by craxal** [comment]
@KerbalSpike I assume you are able to work around this issue by removing the trailing space from the connection string? Do you know how the space got there (error in typing, space was in the connection string already, space was inserted when you copied, etc.)? Either way, we should be able to add this situation to our test cases and handle it appropriately.

**Comment by KerbalSpike** [comment]
@craxal 
Your assumption is of course correct. Once I knew what was causing the issue, it was really simple to fix: just connect with the proper connection string without a trailing space.

I just wanted to make sure to document this issue here, because it had annoyed me for many weeks and searching for the error message brought me nowhere closer to solving it. Funnily enough I only discovered the real reason (the trailing blank) while trying to gather all details to report the issue "unable to delete entities from tables for certain storage accounts".

I don't know how the space got there, but the most probable explanation is that it was accidentally included when I copied the SAS string from somewhere else.


I guess one way to fix this issue would be to do something like calling .Trim() on the connection string before using it? 🤷‍♂️
